test/albumController.test.js

```diff
@@ -0,0 +1,190 @@
+    expect(response.body.albums).toHaveLength(1);
+  });
+
+  it('should return an empty list if there are no albums in the database', async () => {
+    await Album.deleteMany({});
+    // Make a GET request to the /api/v1/albums endpoint
+    const response = await request(app).get('/api/v1/albums');
+
+    // Assert that the response status is OK (200)
+    expect(response.status).toBe(StatusCodes.OK);
+
+    // Assert that the response body is an object with "albums" and "count" keys
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body).toHaveProperty('count');
+
+    // Assert that "albums" is an empty array since there are no albums in the database
+    expect(response.body.albums).toEqual([]);
+
+    // Assert that the "count" field reflects the actual count of albums (which is 0 in this case)
+    expect(response.body.count).toBe(0);
+  });
+
+  it('should test the getSingleAlbum endpoint - Success Case', async () => {
+    const { _id: albumId } = await Album.findOne({});
+
+    const response = await request(app).get(`/api/v1/albums/${albumId}`);
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+  });
+
+  it('should test the getSingleAlbum endpoint - Error Case (Not Found)', async () => {
+    const response = await request(app).get('/api/v1/albums/invalidAlbumId');
+    expect(response.status).toBe(StatusCodes.NOT_FOUND);
+  });
+
+  it('should test the getAlbumWithAllUsersWhoPurchasedIt endpoint - Success Case', async () => {
+    // Try to find an album; if not found, create a default one
+    const album =
+      (await Album.findOne({})) ||
+      (await Album.create({
+        albumName: 'Album',
+        artistName: 'Artist',
+      }));
+
+    const adminCredentials = {
+      email: 'admin@admin.com',
+      password: 'adminpassword',
+    };
+
+    // Use the loginAndReturnCookie function to log in as admin and get the signed cookie
+    const signedCookie = await loginAndReturnCookie(adminCredentials);
+
+    // Make a request to the endpoint and set the obtained cookie
+    const response = await request(app)
+      .get(`/api/v1/albums/${album._id}/listOfUsersWhoPurchasedThisAlbum`)
+      .set('Cookie', signedCookie); // Set the obtained cookie
+
+    // Check if the response status is OK and contains the expected properties
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+    expect(response.body).toHaveProperty('purchasingUsersCount');
+  });
+
+  it('should test the getFilteredAlbums endpoint - Success Case', async () => {
+    // Create 1 album in the database
+    const albumToCreate = {
+      albumName: 'Album 1',
+      artistName: 'Artist 1',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+    };
+
+    await Album.create(albumToCreate);
+
+    const response = await request(app).get(
+      '/api/v1/albums?limit=10&order=asc'
```

Comment:
This is currently testing the `getAllAlbums` endpoint `/api/v1/albums` which doesn't handle query parameters and will just ignore the `limit` and `order` passed here.

The `getFilteredAlbums` route is defined here: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/8f5ec32895b103423ab561869b2479b76223af5d/src/routes/albumRoutes.js#L24](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/8f5ec32895b103423ab561869b2479b76223af5d/src/routes/albumRoutes.js#L24)

so this url should be updated to:

```suggestion
      '/api/v1/albums/filter?limit=10&order=asc'
```

Reply:
Thank you, understood 👍